client/src/main/scala/org/http4s/client/ConnectionFailure.scala

```diff
@@ -0,0 +1,11 @@
+package org.http4s.client
+
+import java.io.IOException
+
+/** Indicates a failure to establish a client connection, preserving the request key
+  * that we tried to connect to.
+  */
+class ConnectionFailure(requestKey: RequestKey, cause: Throwable) extends IOException(cause) {
```

Comment:
This class is not used. Drop?

Reply:
I think the idea is that it will be used in next release of http4s, but @rossabaker chose to use backward-compatible solution for `0.20` release